src/hooks/useKeyboardShortcut/index.ts

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useEffect, useRef } from "react";
+import { CharKey, ModifierKey } from "constants/keys";
+
+type UseKeyboardShortcutOptions = {
+  disabled?: boolean;
+  preventDefault?: boolean;
+  overrideIgnore?: boolean;
+};
+
+// Used to prevent shortcuts from being activated on input elements.
+const INPUT_ELEMENTS = ["INPUT", "TEXTAREA", "SELECT"];
+
+const useKeyboardShortcut = (
+  shortcutKeys: (ModifierKey | CharKey)[] | CharKey,
+  cb: () => void,
+  options: UseKeyboardShortcutOptions = {}
+) => {
+  // We wrap the callback to prevent triggering unnecessary useEffect.
+  const cbRef = useRef(cb);
+  cbRef.current = cb;
+
+  const {
+    disabled = false,
+    preventDefault = true,
+    overrideIgnore = false,
+  } = options;
+
+  const isShortcutPressed = useCallback(
+    (event: KeyboardEvent) => {
+      if (typeof shortcutKeys === "string") {
+        return (
+          shortcutKeys === event.key &&
+          !event.metaKey &&
+          !event.ctrlKey &&
+          !event.shiftKey &&
+          !event.altKey
+        );
+      }
+
+      const shouldControl = shortcutKeys.includes(ModifierKey.Control);
+      if (shouldControl && !event.ctrlKey && !event.metaKey) return false;
+      if (!shouldControl && (event.ctrlKey || event.metaKey)) return false;
+
+      const shouldAlt = shortcutKeys.includes(ModifierKey.Alt);
+      if (shouldAlt && !event.altKey) return false;
+      if (!shouldAlt && event.altKey) return false;
+
+      const shouldShift = shortcutKeys.includes(ModifierKey.Shift);
+      if (shouldShift && !event.shiftKey) return false;
+      if (!shouldShift && event.shiftKey) return false;
```

Comment:
I don't know how to make this method less ugly 😔

The shortcut keys pressed need to be exact, i.e. pressing `Ctrl + Shift + F` should not activate the shortcut for `Ctrl + F`. This is kind of why I preferred the strategy of keeping track of `heldKeys[]`, but as I explained before that will not work